cmd/root.go

```diff
@@ -59,6 +64,12 @@ var RootCmd = &cobra.Command{
 func init() {
 	RootCmd.PersistentFlags().IntVarP(&numRetries, "retries", "r", DefaultNumRetries, "For SSM, the number of retries we'll make before giving up")
 	RootCmd.PersistentFlags().BoolVarP(&verbose, "verbose", "", false, "Print more information to STDOUT")
+	RootCmd.PersistentFlags().StringVarP(&backendFlag, "backend", "b", "ssm",
```

Comment:
seems strange that this value is either an "enum"-like type or an s3 url. how about the valid values are "ssm", "s3", or "null" and then if it is "s3", we require another flag called `--s3-backend-bucket`?

Reply:
hmm yeah i didn't see ssm:// as being a meaningful URL.

i agree requiring both env var and flags will be confusing. Requiring flags for a backend configuration is annoying, but we're already requiring the user to specify a backend override via flags to use s3.

could we recommend users to use aliases for default backends? 
`alias chambers3=chamber --backend s3 --s3-backend-bucket mybucket`

we already recommend this pattern with other use cases (e.g. aws-okta):
`alias chamberprod=aws-okta exec production -- chamber`

Reply:
I would find `--backend s3://mysecrets` easiest to use. Seems like it makes the code inside chamber uglier though. Given the choice between easy for the user and easy for the developer, with chamber, I'd choose easy for the user. It's a relatively small/simple codebase where usability matters a lot.

Recommending some aliases seems like a solid way to improve usability:
```
alias chambers3prod=aws-okta exec prod-privileged -- chamber --backend s3 --s3-backend-bucket prodbucket
alias chambers3stage=aws-okta exec stage-privileged -- chamber --backend s3 --s3-backend-bucket stagebucket
```

Reply:
In terms of env vars vs. command line, I usually assume that command line takes precedence, since it's more tightly scoped

Reply:
here's my stance

1. creating aliases is very easy and fixes any UX issues someone might have by required flags
2. I am interpreting the current backend config  as a string (and not a url). i would expect users to think the same if the values were currently "null" and "ssm"
3. given the current config is not explicitly a url, setting it as a url could cause confusion for a future engineer. if we add a new backend in the future that requires multiple configuration options, we're going to need to add flags (unless we want to somehow overload this setting with more options via some string formatting)

this case isn't that big of an issue. but it might set the standard for future backends. If I were adding this, I'd personally add the additional command line flag, but I won't fight for it if you feel strongly

Reply:
> if we add a new backend in the future that requires multiple configuration options, we're going to need to add flags (unless we want to somehow overload this setting with more options via some string formatting)

I was thinking that if we need options, they'd be query params, like `s3://bucket?retries=5`.

Personally, I like the url syntax because it's so compact vs the extra flags (especially if we grow more flags). Maybe I'm used to it from other apps like `gsutil` and DB connection strings, but I can imagine that being unfamiliar and weird.

Anyway, besides s3, there's really no use case for it at present, so I think I'll just back that out in favour of specific flags. If we get more backends or more options, we can revisit then.

How do y'all feel about `--backend-<backend>-<option>`? i.e. `--backend-s3-bucket` Just so they sort together.

Reply:
`--backend-s3-bucket` sounds good to me

Reply:
Ok, I've removed the URI stuff in the latest.